### PR TITLE
[cxx-interop] Disambiguate const and non-const methods consistently

### PR DESCRIPTION
**Pull request by egorzhdan**
When importing a C++ struct that contains two methods that only differ in const-ness, we append `Mutating` to the name of the non-const method to make it possible to call from Swift unambiguously.

Unfortunately that logic was dependent on the order in which we import methods of a class: the `Mutating` suffix was added when another method with the same name was already imported.

This caused lookup failures, and the behavior was incorrect when the pair of methods return instances of an unsafe type: the const overload was renamed as `Unsafe` properly, but the non-const overload was not renamed.

### CODE REVIEW

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

File: lib/ClangImporter/ImportName.cpp [hunk 1]
```diff
@@ -2200,6 +2200,45 @@ ImportedName NameImporter::importNameImpl(const clang::NamedDecl *D,
...
+        if (otherDecl == D)
+          continue;
+        if (auto otherMethod = dyn_cast<clang::CXXMethodDecl>(otherDecl)) {
+          // TODO: what if the other method is also non-const?
+          if (!otherMethod->isConst())
+            continue;
+          // If the parameters are different, there is no need to disambiguate.
+          if (otherMethod->getNumParams() != method->getNumParams())
+            continue;
+          bool sameParams = llvm::none_of(
```
Comment (hyp) [inline lib/ClangImporter/ImportName.cpp:2221]: I don't think we were comparing parameter types before , were we? why are we comparing them now.

**Comment by hyp** [comment]
Potentially source breaking for some adopters, can you bump up the interop version too in the JSON file?

File: lib/ClangImporter/ImportName.cpp (hunk 1 above)
Comment (egorzhdan) [reply lib/ClangImporter/ImportName.cpp:2221]: You're right, thanks, I assumed for some reason that we were comparing `DeclName`s, but turns out it was just strings. I'll get rid of this check.

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test Windows

**Comment by egorzhdan** [comment]
@swift-ci please smoke test

**Pull request merged**